[09/01/23 18:28:35] INFO     colossalai - colossalai - INFO:                    
                             /scratch/yl9315/miniconda3/envs/coati1/lib/python3.
                             10/site-packages/colossalai/context/parallel_contex
                             t.py:522 set_device                                
                    INFO     colossalai - colossalai - INFO: process rank 0 is  
                             bound to device 0                                  
[09/01/23 18:28:36] INFO     colossalai - colossalai - INFO:                    
                             /scratch/yl9315/miniconda3/envs/coati1/lib/python3.
                             10/site-packages/colossalai/context/parallel_contex
                             t.py:558 set_seed                                  
                    INFO     colossalai - colossalai - INFO: initialized seed on
                             rank 0, numpy: 42, python random: 42,              
                             ParallelMode.DATA: 42, ParallelMode.TENSOR: 42,the 
                             default parallel seed is ParallelMode.DATA.        
                    INFO     colossalai - colossalai - INFO:                    
                             /scratch/yl9315/miniconda3/envs/coati1/lib/python3.
                             10/site-packages/colossalai/initialize.py:115      
                             launch                                             
                    INFO     colossalai - colossalai - INFO: Distributed        
                             environment is initialized, data parallel size: 1, 
                             pipeline parallel size: 1, tensor parallel size: 1 
/scratch/yl9315/miniconda3/envs/coati1/lib/python3.10/site-packages/colossalai/booster/booster.py:70: UserWarning: The plugin will control the accelerator, so the device argument will be ignored.
  warnings.warn('The plugin will control the accelerator, so the device argument will be ignored.')
/scratch/yl9315/miniconda3/envs/coati1/lib/python3.10/site-packages/colossalai/booster/booster.py:76: UserWarning: The plugin will control the precision, so the mixed_precision argument will be ignored.
  warnings.warn('The plugin will control the precision, so the mixed_precision argument will be ignored.')
/scratch/yl9315/miniconda3/envs/coati1/lib/python3.10/site-packages/colossalai/kernel/op_builder/utils.py:94: UserWarning: [extension] The CUDA version on the system (11.6) does not match with the version (11.7) torch was compiled with. The mismatch is found in the minor version. As the APIs are compatible, we will allow compilation to proceed. If you encounter any issue when using the built kernel, please try to build it again with fully matched CUDA versions
  warnings.warn(
/scratch/yl9315/miniconda3/envs/coati1/lib/python3.10/site-packages/colossalai/kernel/op_builder/utils.py:94: UserWarning: [extension] The CUDA version on the system (11.6) does not match with the version (11.7) torch was compiled with. The mismatch is found in the minor version. As the APIs are compatible, we will allow compilation to proceed. If you encounter any issue when using the built kernel, please try to build it again with fully matched CUDA versions
  warnings.warn(
[09/01/23 18:29:38] INFO     colossalai - colossalai - INFO:                    
                             /scratch/yl9315/miniconda3/envs/coati1/lib/python3.
                             10/site-packages/coati/dataset/prompt_dataset.py:24
                              __init__                                          
                    INFO     colossalai - colossalai - INFO: Loading data...    
                    INFO     colossalai - colossalai - INFO:                    
                             /scratch/yl9315/miniconda3/envs/coati1/lib/python3.
                             10/site-packages/coati/dataset/prompt_dataset.py:26
                              __init__                                          
                    INFO     colossalai - colossalai - INFO: Loaded 283         
                             examples.                                          
                    INFO     colossalai - colossalai - INFO:                    
                             /scratch/yl9315/miniconda3/envs/coati1/lib/python3.
                             10/site-packages/coati/dataset/prompt_dataset.py:29
                              __init__                                          
                    INFO     colossalai - colossalai - INFO: Limiting dataset to
                             16384 examples.                                    
[09/01/23 18:29:39] INFO     colossalai - colossalai - INFO:                    
                             /scratch/yl9315/miniconda3/envs/coati1/lib/python3.
                             10/site-packages/coati/dataset/sft_dataset.py:120  
                             __init__                                           
                    INFO     colossalai - colossalai - INFO: Loading data...    
                    INFO     colossalai - colossalai - INFO:                    
                             /scratch/yl9315/miniconda3/envs/coati1/lib/python3.
                             10/site-packages/coati/dataset/sft_dataset.py:122  
                             __init__                                           
                    INFO     colossalai - colossalai - INFO: Loaded 765         
                             examples.                                          
                    INFO     colossalai - colossalai - INFO:                    
                             /scratch/yl9315/miniconda3/envs/coati1/lib/python3.
                             10/site-packages/coati/dataset/sft_dataset.py:125  
                             __init__                                           
                    INFO     colossalai - colossalai - INFO: Limiting dataset to
                             16384 examples.                                    
                    INFO     colossalai - colossalai - INFO:                    
                             /scratch/yl9315/miniconda3/envs/coati1/lib/python3.
                             10/site-packages/coati/dataset/sft_dataset.py:128  
                             __init__                                           
                    INFO     colossalai - colossalai - INFO: Formatting         
                             inputs...                                          
                    INFO     colossalai - colossalai - INFO:                    
                             /scratch/yl9315/miniconda3/envs/coati1/lib/python3.
                             10/site-packages/coati/dataset/sft_dataset.py:139  
                             __init__                                           
                    INFO     colossalai - colossalai - INFO: Tokenizing         
                             inputs... This may take some time...               
Episodes:   0%|          | 0/40 [00:00<?, ?it/s]
Collect steps:   0%|          | 0/10 [00:00<?, ?it/s][ACollect steps:   0%|          | 0/10 [01:06<?, ?it/s]
Episodes:   0%|          | 0/40 [01:06<?, ?it/s]
Traceback (most recent call last):
  File "/scratch/yl9315/./train_prompts_1.py", line 233, in <module>
    main(args)
  File "/scratch/yl9315/./train_prompts_1.py", line 189, in main
    trainer.fit(prompt_dataloader=prompt_dataloader,
  File "/scratch/yl9315/miniconda3/envs/coati1/lib/python3.10/site-packages/coati/trainer/base.py", line 179, in fit
    self._collect_phase(collect_step)
  File "/scratch/yl9315/miniconda3/envs/coati1/lib/python3.10/site-packages/coati/trainer/base.py", line 145, in _collect_phase
    experience = self._make_experience(collect_step)
  File "/scratch/yl9315/miniconda3/envs/coati1/lib/python3.10/site-packages/coati/trainer/ppo.py", line 122, in _make_experience
    return self.experience_maker.make_experience(**prompts, **self.generate_kwargs)
  File "/scratch/yl9315/miniconda3/envs/coati1/lib/python3.10/site-packages/torch/autograd/grad_mode.py", line 27, in decorate_context
    return func(*args, **kwargs)
  File "/scratch/yl9315/miniconda3/envs/coati1/lib/python3.10/site-packages/coati/experience_maker/naive.py", line 49, in make_experience
    r = self.reward_model(sequences, attention_mask, action_mask, num_actions)
  File "/scratch/yl9315/miniconda3/envs/coati1/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1194, in _call_impl
    return forward_call(*input, **kwargs)
  File "/scratch/yl9315/miniconda3/envs/coati1/lib/python3.10/site-packages/colossalai/booster/plugin/low_level_zero_plugin.py", line 166, in forward
    return super().forward(*args, **kwargs)
  File "/scratch/yl9315/miniconda3/envs/coati1/lib/python3.10/site-packages/colossalai/interface/model.py", line 25, in forward
    return self.module(*args, **kwargs)
  File "/scratch/yl9315/miniconda3/envs/coati1/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1194, in _call_impl
    return forward_call(*input, **kwargs)
  File "/scratch/yl9315/reward_model/ratio_rm.py", line 34, in forward
    score = self.compute_reward(prob)
  File "/scratch/yl9315/reward_model/ratio_rm.py", line 19, in compute_reward
    return torch.matmul(prob,self.score_matrix)
RuntimeError: size mismatch, got 8192, 8192x5,4
ERROR:torch.distributed.elastic.multiprocessing.api:failed (exitcode: 1) local_rank: 0 (pid: 3265475) of binary: /scratch/yl9315/miniconda3/envs/coati1/bin/python
Traceback (most recent call last):
  File "/scratch/yl9315/miniconda3/envs/coati1/bin/torchrun", line 8, in <module>
    sys.exit(main())
  File "/scratch/yl9315/miniconda3/envs/coati1/lib/python3.10/site-packages/torch/distributed/elastic/multiprocessing/errors/__init__.py", line 346, in wrapper
    return f(*args, **kwargs)
  File "/scratch/yl9315/miniconda3/envs/coati1/lib/python3.10/site-packages/torch/distributed/run.py", line 762, in main
    run(args)
  File "/scratch/yl9315/miniconda3/envs/coati1/lib/python3.10/site-packages/torch/distributed/run.py", line 753, in run
    elastic_launch(
  File "/scratch/yl9315/miniconda3/envs/coati1/lib/python3.10/site-packages/torch/distributed/launcher/api.py", line 132, in __call__
    return launch_agent(self._config, self._entrypoint, list(args))
  File "/scratch/yl9315/miniconda3/envs/coati1/lib/python3.10/site-packages/torch/distributed/launcher/api.py", line 246, in launch_agent
    raise ChildFailedError(
torch.distributed.elastic.multiprocessing.errors.ChildFailedError: 
============================================================
./train_prompts_1.py FAILED
------------------------------------------------------------
Failures:
  <NO_OTHER_FAILURES>
------------------------------------------------------------
Root Cause (first observed failure):
[0]:
  time      : 2023-09-01_18:30:50
  host      : ga002.hpc.nyu.edu
  rank      : 0 (local_rank: 0)
  exitcode  : 1 (pid: 3265475)
  error_file: <N/A>
  traceback : To enable traceback see: https://pytorch.org/docs/stable/elastic/errors.html
============================================================
